File: 268.missing-number.1.go
/*
 * @lc app=leetcode id=268 lang=golang
 *
 * [268] Missing Number
 */
func missingNumber(nums []int) int {
    sum := 0
    for i, n := range nums {
        sum += i - n
    }
    return sum + len(nums)
}

File: 268.missing-number.1.rs
/*
 * @lc app=leetcode id=268 lang=rust
 *
 * [268] Missing Number
 */
impl Solution {
    pub fn missing_number(nums: Vec<i32>) -> i32 {
        let mut sum = 0;
        for (i, n) in nums.iter().enumerate() {
            sum += (i as i32) - n;
        }
        return sum + (nums.len() as i32)
    }
}
